server/routes/documents/letters-list/letters-list.controller.js

```diff
@@ -264,14 +265,40 @@
   // the same approach could be used in other places where we have checkboxes for selecting items from big lists
   module.exports.checkJuror = function(app) {
     return function(req, res) {
-      const { isChecking } = req.body;
+      const { isChecking, isCheckAll } = req.body;
 
-      if (!req.session.documentsJurorsList.checkedJurors) {
-        req.session.documentsJurorsList.checkedJurors = [];
+      if (isCheckAll === 'true') {
+        if (areAllChecked(req)) {
+          req.session.documentsJurorsList.checkedJurors = [];
+        } else {
+          req.session.documentsJurorsList.checkedJurors = [];
+          req.session.documentsJurorsList.data.forEach(juror => {
+            if (!juror[4] && !juror[5]) {
+              req.session.documentsJurorsList.checkedJurors.push({
+                'juror_number': juror[0],
+                'form_code': juror[6],
+                'date_printed': req.body.date_printed,
+              });
+            }
+          });
+        }
+
+        app.logger.info('Checked / unchecked all juror documents: ', {
+          auth: req.session.authentication,
+          jwt: req.session.authToken,
```

Comment:
we should not put jwt in any of the new logging objects

Reply:
Removed :)